workflow/executor/os-specific/command_test.go

```diff
@@ -34,9 +34,9 @@ func TestSimpleStartCloser(t *testing.T) {
 	time.Sleep(100 * time.Millisecond)
 	closer()
 
-	expected := "A123456789B123456789C123456789D123456789E123456789"
+	expected := "A123456789B123456789C123456789D123456789E123456789\n"
 	if runtime.GOOS == "windows" {
-		expected = "A123456789B123456789C123456789D123456789E123456789\\c\r\n"
+		expected = "A123456789B123456789C123456789D123456789E123456789\r\n"
```

Comment:
It is difficult to determine whether the `echo` command used by the current OS supports `\c`.
I also considered the following two options:
1. using `echo -e` to explicitly enable escape characters, problems:
- `pwsh` do not support `-e`
- `sh` of old versions do not support `-e`
2. using `echo -n` instead of `\c`, problems:
- `pwsh` do not support `-n`

So in the end I think it's better to just keep the newline.

Reply:
My problem with changing this is that  both of these tests it that the original author specifically added a `\c` to the code.

This implies they wanted to test this case, and changing the test case for all the platforms feels like the wrong approach.

If it doesn't work for you in your native environment can you run inside the devcontainer where it has a shell that works with this?

Reply:
After checking the context, I didn't see the need to test `\c` here, and it is clearly that this would not take effect on CentosOS or Windows.

> If it doesn't work for you in your native environment can you run inside the devcontainer where it has a shell that works with this?

I just wish `make test` works for all platforms to simplify development.

Reply:
The original PR with `\c` replaced the `-n` due to test failures: https://github.com/argoproj/argo-workflows/pull/11576

Reply:
So my take is that #11368 is about ensuring that all of stdout is captured, and the tests are trying to prove that.

Stdout is buffered and connected to a terminal, so will only be flushed on a new line. Reference:
https://www.pixelbeat.org/programming/stdio_buffering/

So by putting the new line in to the echo we'll end up with a test which no longer tests that buffered output which hasn't been flushed is captured by the system.

The closer introduced in #11368 is there to force flush stdout, and these tests are there to create buffered unflushed output and prove we capture it.

Reply:
Thanks for the analysis Alan! We should definitely add a comment to mention that the newline is intentionally there for testing buffering / flushing 😅 

In that case, I would suggest detecting CentOS and having a different behavior for it, since this seems to affects variants / derivatives of CentOS https://github.com/argoproj/argo-workflows/discussions/13565

Reply:
I don't get it. #11368 add `cmd.Wait()` to ensure the command completed and all output was written to stdout, why does it matter with '\n'？
In my opinion, this test just checks whether `hello` in stdout is successfully captured. Why don't we just check if the output is consistent with `hello\n` instead of investigating the OS suitability of `-n` and `\c`, which have been proven to be applicable to different OSes.

Reply:
So I went down this rabbit hole today, is it not bad that the delay we give to copy IO before erroring out is just a hard coded 100ms?